geoserve/geoserve.go

```diff
@@ -101,64 +99,57 @@ func (server *GeoServer) run() {
 	for {
 		select {
 		case g := <-server.cacheGet:
-			var jsonData []byte
 			_jsonData, found := server.cache.Get(g.ip)
 			if found {
 				log.Trace("Cache hit")
-				jsonData = _jsonData.([]byte)
-			} else {
-				log.Trace("Cache miss, looking up geolocation info")
-				geoData, err := server.db.City(net.ParseIP(g.ip))
-				if err != nil {
-					log.Errorf("Unable to look up ip address %s: %s", g.ip, err)
-				} else {
-					jsonData, err = json.Marshal(geoData)
-					if err != nil {
-						log.Errorf("Unable to encode json response for ip address: %s", g.ip)
-					} else {
-						// Cache it
-						server.cache.Add(g.ip, jsonData)
-					}
-				}
+				g.resp <- _jsonData.([]byte)
```

Comment:
If we had a cache hit, I assume we should stop process and not call lookupDB, right?

Reply:
Ahh dumb error. Meant to `continue`